RecoTauTag/RecoTau/plugins/PATTauDiscriminationByMVAIsolationRun2.cc

```diff
@@ -0,0 +1,306 @@
+
+/*
+ * \class PATTauDiscriminationByMVAIsolationRun2
+ * 
+ * MVA based discriminator against jet -> tau fakes
+ * 
+ * Adopted from RecoTauTag/RecoTau/plugins/PFRecoTauDiscriminationByMVAIsolationRun2.cc
+ * to enable computation of MVA isolation on MiniAOD
+ * 
+ * \author Alexander Nehrkorn, RWTH Aachen
+ */
+
+// todo 1: remove leadingTrackChi2 as input variable from:
+//           - here
+//           - TauPFEssential
+//           - PFRecoTauDiscriminationByMVAIsolationRun2
+//           - Training of BDT
+// todo 2: do we need/want to add PATTauIDEmbedder?
+
+#include "RecoTauTag/RecoTau/interface/TauDiscriminationProducerBase.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+
+#include "FWCore/Utilities/interface/Exception.h"
+
+#include "DataFormats/Candidate/interface/Candidate.h"
+#include "DataFormats/PatCandidates/interface/Tau.h"
+#include "DataFormats/PatCandidates/interface/PATTauDiscriminator.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "RecoTauTag/RecoTau/interface/PFRecoTauClusterVariables.h"
+
+#include "CondFormats/EgammaObjects/interface/GBRForest.h"
+#include "CondFormats/DataRecord/interface/GBRWrapperRcd.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include <TMath.h>
+#include <TFile.h>
+
+#include <iostream>
+
+using namespace pat;
+
+namespace
+{
+  const GBRForest* loadMVAfromFile(const edm::FileInPath& inputFileName, const std::string& mvaName, std::vector<TFile*>& inputFilesToDelete)
+  {
+    if ( inputFileName.location() == edm::FileInPath::Unknown ) throw cms::Exception("PATTauDiscriminationByIsolationMVARun2::loadMVA")
+      << " Failed to find File = " << inputFileName << " !!\n"; 
+    TFile* inputFile = new TFile(inputFileName.fullPath().data());
```

Comment:
I will check -- the impacts on multi-threaded running is not clear to me. I also do not know how I should check this (you can let me know if you want me to do anythign about it). But I can say that this structure is a clone of an analogue for reco::PFTau's, so the technology is well established and used within official reco sequences. This one as you stated properly below will not enter official sequences. It is meant for the use on already available miniAOD content for final analysis skims.